src/bus/spi/mod.rs

```diff
@@ -1,10 +1,9 @@
-pub mod bus;
-pub mod read;
-pub mod write;
-
 #[cfg(not(feature = "fugit"))]
 use core::time::Duration;
 
+use embedded_hal_async::spi::{ErrorType, SpiDevice, SpiBus};
+use log::trace;
```

Comment:
we could add either `log` or `defmt` as features to allow either one of the crates to be used.

Reply:
You mean let user choose which crate to use?

Reply:
Yes, some like e.g.:

https://github.com/embassy-rs/embassy/blob/f9c0c53e126e845f202d56f9402b61c45b05c87a/embassy-rp/Cargo.toml#L51-L52

https://github.com/embassy-rs/embassy/blob/5e94b8060b7f1af8118ca1f943181ce70be92057/embassy-boot/rp/src/fmt.rs#L132-L143

Also do not allow  both feature to be enabled at the same time:

https://github.com/embassy-rs/embassy/blob/master/embassy-net-driver-channel/src/fmt.rs#L4-L5

but I've also seen features that only switch between `use log::...` & `use defmt::...`, the only to be careful is running checks for both features builds because `defmt` requires `derive(Format)` or special attributes for values you use in log macros.

Reply:
It's ok to me, and not in this issue and PR :)

Reply:
Great, and I agree.